### PR TITLE
Adding configuration files to support both 1.1.6 and 1.1.8

### PR DESCRIPTION
**Pull request by cmgerdes**
I've done my best to clean printer-specific configuration from the files, this should work with most default VORONs.  Someone should come in behind me and double check my work though.

### CODE REVIEW

**Comment by Fulg** [comment]
Thank you for sharing this, I'll have a look at it soon.

File: Firmware/Marlin/1.1.6/Configuration.h [hunk 1]
```diff
@@ -0,0 +1,1709 @@
...
+/**
+ * This setting determines the communication speed of the printer.
+ *
+ * 250000 works in most cases, but you might try a lower speed if
+ * you commonly experience drop-outs during host printing.
+ * You may try up to 1000000 to speed up SD file transfer.
+ *
+ * :[2400, 9600, 19200, 38400, 57600, 115200, 250000, 500000, 1000000]
+ */
+#define BAUDRATE 250000
```
Comment (Fulg) [inline Firmware/Marlin/1.1.6/Configuration.h]: Default baud rate should be 115200 on VORON (like the original).

File: Firmware/Marlin/1.1.6/Configuration.h [hunk 2]
```diff
@@ -0,0 +1,1709 @@
...
+// If this is enabled, find your own PID constants below.
+#define PIDTEMPBED
+
+//#define BED_LIMIT_SWITCHING
+
+// This sets the max power delivered to the bed, and replaces the HEATER_BED_DUTY_CYCLE_DIVIDER option.
+// all forms of bed control obey this (PID, bang-bang, bang-bang with hysteresis)
+// setting this to anything other than 255 enables a form of PWM to the bed just like HEATER_BED_DUTY_CYCLE_DIVIDER did,
+// so you shouldn't use it unless you are OK with PWM on your bed.  (see the comment on enabling PIDTEMPBED)
+#define MAX_BED_POWER 225 // limits duty cycle to bed; 255=full current
```
Comment (Fulg) [inline Firmware/Marlin/1.1.6/Configuration.h]: `MAX_BED_POWER` should be 205 (80%).

**Comment by Fulg** [comment]
- I don't think the 1.1.6 folder is useful (it didn't work for you!), I would just add 1.1.8.
- You should move the existing 1.0-RC8 files in their own folder.

File: Firmware/Marlin/1.1.8/pins_RAMPS_14_VORON.h [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ * Arduino Mega with RAMPS v1.4 pin assignments
+ *
+ *  RAMPS_14_VORON (VORON pin config for RAMPs 1.4)
+ */
+
+// Include the base configuration for RAMPs 1.4
+#include "pins_RAMPS.h"
+
+#define CONTROLLER_FAN_PIN  -1 // Changed to -1 since hooked to always on.
```
Comment (Fulg) [inline Firmware/Marlin/1.1.8/pins_RAMPS_14_VORON.h]: `CONTROLLER_FAN_PIN` should be 8 as per stock wiring diagrams.

File: Firmware/Marlin/1.1.8/pins_RAMPS_14_VORON.h [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ 							 // LED strip can be attached to the same output. This way LEDs are only on during printing, and shortly afteR
+//#define USE_CONTROLLER_FAN true
+
+// Fan Expansion board pins
+#define E0_AUTO_FAN_PIN 6 // Servo pin 6 for E3D Fan
+#define E1_AUTO_FAN_PIN 6 // Servo pin 6 for E3D Fan (same pin for both extruders since it's the same fan)
+
+#define FAN_PIN            		8 // Using the pin for the controller fan since controller fan is always on.
```
Comment (Fulg) [inline Firmware/Marlin/1.1.8/pins_RAMPS_14_VORON.h]: `FAN_PIN` should be 5 per stock wiring diagrams.

File: Firmware/Marlin/1.1.8/ultralcd.cpp [hunk 5]
```diff
@@ -0,0 +1,5158 @@
...
+  /**
+   *
+   * "Main" menu
+   *
+   */
+
+  void lcd_main_menu() {
+    START_MENU();
+    MENU_BACK(MSG_WATCH);
+
```
Comment (Fulg) [inline Firmware/Marlin/1.1.8/ultralcd.cpp]: Missing:

```
    MENU_ITEM(function, "Lower Bed", lcd_lowerbed);
```

File: Firmware/Marlin/1.1.8/ultralcd.cpp [hunk 6]
```diff
@@ -0,0 +1,5158 @@
...
+        char msg[10];
+        sprintf_P(msg, PSTR("%i / %u"), (int)(manual_probe_index + 1), total_probe_points);
+        lcd_implementation_drawedit(PSTR(MSG_LEVEL_BED_NEXT_POINT), msg);
+      }
+      lcdDrawUpdate = LCDVIEW_CALL_NO_REDRAW;
+      if (!lcd_wait_for_move) lcd_goto_screen(_lcd_level_bed_get_z);
+    }
+
+    /**
+     * Step 5: I
```
Comment (Fulg) [inline Firmware/Marlin/1.1.8/ultralcd.cpp]: Missing:

```
      MENU_ITEM(function, "Re-home Z", lcd_rehome_z);
```

Comment (Fulg) [reply Firmware/Marlin/1.1.8/ultralcd.cpp]: I don't know why the preview messed up, I meant to add this after line 3598, as in:
```
void lcd_control_motion_menu() {
  START_MENU();
  MENU_BACK(MSG_CONTROL);

  #if ENABLED(BABYSTEP_ZPROBE_OFFSET)
    MENU_ITEM(submenu, MSG_ZPROBE_ZOFFSET, lcd_babystep_zoffset);
  #elif HAS_BED_PROBE
    MENU_ITEM_EDIT(float32, MSG_ZPROBE_ZOFFSET, &zprobe_zoffset, Z_PROBE_OFFSET_RANGE_MIN, Z_PROBE_OFFSET_RANGE_MAX);
      MENU_ITEM(function, "Re-home Z", lcd_rehome_z);
  #endif
```

File: Firmware/Marlin/1.1.8/Configuration.h [hunk 7]
```diff
@@ -0,0 +1,1791 @@
...
+/**
+ * This setting determines the communication speed of the printer.
+ *
+ * 250000 works in most cases, but you might try a lower speed if
+ * you commonly experience drop-outs during host printing.
+ * You may try up to 1000000 to speed up SD file transfer.
+ *
+ * :[2400, 9600, 19200, 38400, 57600, 115200, 250000, 500000, 1000000]
+ */
+#define BAUDRATE 250000
```
[the remark this answers is not in the document]
Comment (putertdi) [reply Firmware/Marlin/1.1.8/Configuration.h]: Fixed - this was also most likely what was causing the issues I was seeing.

File: Firmware/Marlin/1.1.8/Configuration.h [hunk 8]
```diff
@@ -0,0 +1,1791 @@
...
+// If this is enabled, find your own PID constants below.
+#define PIDTEMPBED
+
+//#define BED_LIMIT_SWITCHING
+
+// This sets the max power delivered to the bed, and replaces the HEATER_BED_DUTY_CYCLE_DIVIDER option.
+// all forms of bed control obey this (PID, bang-bang, bang-bang with hysteresis)
+// setting this to anything other than 255 enables a form of PWM to the bed just like HEATER_BED_DUTY_CYCLE_DIVIDER did,
+// so you shouldn't use it unless you are OK with PWM on your bed.  (see the comment on enabling PIDTEMPBED)
+#define MAX_BED_POWER 225 // limits duty cycle to bed; 255=full current
```
[the remark this answers is not in the document]
Comment (putertdi) [reply Firmware/Marlin/1.1.8/Configuration.h]: updated to 205

File: Firmware/Marlin/1.1.8/Configuration.h [hunk 9]
```diff
@@ -0,0 +1,1791 @@
...
+ * total number of extruders, the last value applies to the rest.
+ */
+#define DISTINCT_E_FACTORS
+
+/**
+ * Default Axis Steps Per Unit (steps/mm)
+ * Override with M92
+ *                                      X, Y, Z, E0 [, E1[, E2[, E3[, E4]]]]
+ */
+#define DEFAULT_AXIS_STEPS_PER_UNIT   { 80, 80, 400, 555.76 }
```
[the remark this answers is not in the document]
Comment (putertdi) [reply Firmware/Marlin/1.1.8/Configuration.h]: switched to 608.  I guess mine ended up being way off.

File: Firmware/Marlin/1.1.8/Configuration.h [hunk 10]
```diff
@@ -0,0 +1,1791 @@
...
+ *    T |           | H
+ *      |    (-)    | T
+ *      |           |
+ *      O-- FRONT --+
+ *    (0,0)
+ */
+#define X_PROBE_OFFSET_FROM_EXTRUDER 0  // X offset: -left  +right  [of the nozzle]
+#define Y_PROBE_OFFSET_FROM_EXTRUDER 0  // Y offset: -front +behind [the nozzle]
+#define Z_PROBE_OFFSET_FROM_EXTRUDER 0   // Z offset: -below +above  [the nozzle]
+
```
[the remark this answers is not in the document]
Comment (putertdi) [reply Firmware/Marlin/1.1.8/Configuration.h]: The strange thing is that I thought I had this in here.  I've added this.

File: Firmware/Marlin/1.1.8/Configuration.h [hunk 11]
```diff
@@ -0,0 +1,1791 @@
...
+ */
+#define Z_CLEARANCE_DEPLOY_PROBE   0 // Z Clearance for Deploy/Stow
+#define Z_CLEARANCE_BETWEEN_PROBES  3 // Z Clearance between probe points
+
+// For M851 give a range for adjusting the Z probe offset
+#define Z_PROBE_OFFSET_RANGE_MIN -20
+#define Z_PROBE_OFFSET_RANGE_MAX 20
+
+// Enable the M48 repeatability test to test probe accuracy
+//#define Z_MIN_PROBE_REPEATABILITY_TEST
```
[the remark this answers is not in the document]
Comment (putertdi) [reply Firmware/Marlin/1.1.8/Configuration.h]: enabled.

File: Firmware/Marlin/1.1.8/Configuration.h [hunk 12]
```diff
@@ -0,0 +1,1791 @@
...
+ *   Probe a grid manually
+ *   The result is a mesh, suitable for large or uneven beds. (See BILINEAR.)
+ *   For machines without a probe, Mesh Bed Leveling provides a method to perform
+ *   leveling in steps so you can manually adjust the Z height at each grid-point.
+ *   With an LCD controller the process is guided step-by-step.
+ */
+//#define AUTO_BED_LEVELING_3POINT
+//#define AUTO_BED_LEVELING_LINEAR
+//#define AUTO_BED_LEVELING_BILINEAR
+#define AUTO_BED_LEVELING_UBL
```
[the remark this answers is not in the document]
Comment (putertdi) [reply Firmware/Marlin/1.1.8/Configuration.h]: It seems to be the recommended way to go at this point.  I'll leave it unless I hear different.

File: Firmware/Marlin/1.1.8/pins_RAMPS_14_VORON.h (hunk 3 above)
Comment (putertdi) [reply Firmware/Marlin/1.1.8/pins_RAMPS_14_VORON.h]: oops, ya, a customization I missed.  There were no breakout boards available when I built my machine.

File: Firmware/Marlin/1.1.8/pins_RAMPS_14_VORON.h (hunk 4 above)
Comment (putertdi) [reply Firmware/Marlin/1.1.8/pins_RAMPS_14_VORON.h]: Fixed

File: Firmware/Marlin/1.1.8/ultralcd.cpp (hunk 5 above)
Comment (putertdi) [reply Firmware/Marlin/1.1.8/ultralcd.cpp]: added.

File: Firmware/Marlin/1.1.8/ultralcd.cpp (hunk 6 above)
Comment (putertdi) [reply Firmware/Marlin/1.1.8/ultralcd.cpp]: Added

File: Firmware/Marlin/1.1.8/ultralcd.cpp [hunk 13]
```diff
@@ -0,0 +1,5158 @@
...
+      //
+      // ^ Main
+      //
+      MENU_BACK(MSG_MAIN);
+      MENU_ITEM_EDIT_CALLBACK(int8, MSG_CASE_LIGHT_BRIGHTNESS, &case_light_brightness, 0, 255, update_case_light, true);
+      MENU_ITEM_EDIT_CALLBACK(bool, MSG_CASE_LIGHT, (bool*)&case_light_on, update_case_light);
+      END_MENU();
+    }
+  #endif // MENU_ITEM_CASE_LIGHT
+
```
Comment (Fulg) [inline Firmware/Marlin/1.1.8/ultralcd.cpp]: It seems GitHub ate one of my review comments, for my other changes to this file to compile you need to add this here:

```
  // VORON Specific tweaks
  void lcd_lowerbed() {
    enqueue_and_echo_commands_P(PSTR("G91\nG1 Z20 F300\nG90"));
  }

  void lcd_rehome_z() {
    enqueue_and_echo_commands_P(PSTR("G28 Z\nG1 Z0 F100"));
  }
```

Comment (putertdi) [reply Firmware/Marlin/1.1.8/ultralcd.cpp]: I went for a walk and came back to the compile error :)  I saw your comment come in on email right as I started to investigate - thanks!

**Comment by putertdi** [comment]
heads up: I have all the files updated but my the thermristor on my second hotend died so it will be a bit before I can finish testing and push the last of the changes.

**Comment by Fulg** [comment]
You should commit what you have now already, nobody can see your local changes yet.

**Comment by putertdi** [comment]
Fair enough, I've committed the changes.  I'm hoping to test them today after I finish repairs.

**Comment by Fulg** [comment]
I think all that's missing is the `lcd_lowerbed()` and `lcd_rehome_z()` functions in ultralcd.cpp, I'm surprised this compiles without it. The rest looks good to me.

**Comment by putertdi** [comment]
It probably doesn't.  I'm having to maintain two version of the file because I have a different config.  I probably updated the one for my printer by mistake - sorry about that.  I'll fix it as soon as I get home today.

**Comment by putertdi** [comment]
Ok, I think this last push should have everything.

Question - after this pull request is approved I have some improvements I'd like to make specifically surrounding how the print bed is probed.  Do you think you or RCF would be open to it?  Specifically, with the spring bed design you get more accurate offsets if you probe further in from the edges so I was going to alter the G29 parameters so that they probe further in so the bed flexes less.

I'm going to do it either way on mine, but I'd like to deviate from the default voron config and push this change out as a suggested change for the printer as a whole.

**Comment by Fulg** [comment]
This PR gets a +1 from me, but the final say is with @mzbotreprap :)

Thanks for putting this together.

As for the bed probing accuracy, I think this really depends on how you built it and which springs you used (for example my bed springs are way too hard to compress from probing, the X/Y rods were bending instead!). But this could be added to the fine-tuning section of the Wiki or something like that.

**Comment by mzbotreprap** [comment]
Nice work guys!

**Pull request merged**